packages/flame/lib/src/effects/provider_interfaces.dart

```diff
@@ -50,3 +50,12 @@ abstract class SizeProvider {
   Vector2 get size;
   set size(Vector2 value);
 }
+
+/// Interface for a component that can be affected by opacity effects.
+abstract class OpacityProvider {
+  double get opacity;
+  set opacity(double value);
+
+  int get alpha;
+  set alpha(int value);
```

Comment:
They are slightly different representation of the same idea. `alpha` is an integer in the range [0-255] and opacity is a double in the range [0-1]. I added both of them just to avoid making changes to `OpacityEffect`'s calculations as they use alpha right now.

Reply:
I've updated the logic of `OpacityEffect` to use opacities instead, in PR #2064.

Reply:
Cool! I'll update mine once that PR is merged.